### PR TITLE
Support to run in parallel some tests [15290]

### PR DESCRIPTION
**Pull request by richiware**
## Description

As title says.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- *N/A* Documentation builds and tests pass locally. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by richiware** [comment]
@richiprosima  Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

File: test/blackbox/common/DDSBlackboxTestsPersistenceGuid.cpp [hunk 1]
```diff
@@ -21,6 +21,9 @@
 
 #include <cstring>
 #include <thread>
+#include <sstream>
+#include <iomanip>
+#include <fstream>
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsPersistenceGuid.cpp:26]: Include what you use: missing \<functional\> header include. Also, sort includes following the style guide.

File: test/unittest/dds/subscriber/DataReaderTests.cpp [hunk 2]
```diff
@@ -64,9 +65,20 @@
 #include <fastdds/rtps/transport/test_UDPv4TransportDescriptor.h>
 #include <fastrtps/xmlparser/XMLProfileManager.h>
 
+#include <asio.hpp>
+
+#if defined(_WIN32)
```
Comment (jsantiago-eProsima) [inline test/unittest/dds/subscriber/DataReaderTests.cpp:70]: Same comment as in the other file. Maybe either add the winrt check to keep it in line with what SystemInfo uses or use that one instead.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: test/unittest/rtps/persistence/PersistenceTests.cpp [hunk 3]
```diff
@@ -24,8 +24,15 @@
...
 #include <climits>
+#include <sstream>
 #include <gtest/gtest.h>
 
+#if defined(_WIN32)
+#define GET_PID _getpid
+#else
+#define GET_PID getpid
+#endif // if defined(_WIN32)
+
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/unittest/rtps/persistence/PersistenceTests.cpp]: Done in d68b310

File: test/blackbox/common/RTPSBlackboxTestsPersistenceGuid.cpp [hunk 4]
```diff
@@ -18,13 +18,21 @@
...
 #include "RTPSWithRegistrationReader.hpp"
 
 #include <gtest/gtest.h>
 #include <fastrtps/xmlparser/XMLProfileManager.h>
 
+#if defined(_WIN32)
+#define GET_PID _getpid
+#else
+#define GET_PID getpid
+#endif // if defined(_WIN32)
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/blackbox/common/RTPSBlackboxTestsPersistenceGuid.cpp]: Done in d68b310

File: test/blackbox/common/DDSBlackboxTestsPersistenceGuid.cpp (hunk 1 above)
Comment (richiware) [reply test/blackbox/common/DDSBlackboxTestsPersistenceGuid.cpp]: Done in d68b310

File: test/unittest/dds/subscriber/DataReaderTests.cpp (hunk 2 above)
Comment (richiware) [reply test/unittest/dds/subscriber/DataReaderTests.cpp]: Done in d68b310

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by jsantiago-eProsima** [comment]
@richiprosima Please test this. I uploaded new configuration to actually test with parallel execution enabled.

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test windows

**Comment by richiware** [comment]
@richiprosima Please test linux

**Comment by richiware** [comment]
@richiprosima Please test linux

**Comment by richiware** [comment]
@richiprosima Please test linux

**Comment by richiware** [comment]
@richiprosima Please test linux

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima  Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima  Please test this

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test this

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test this

**Comment by jsantiago-eProsima** [comment]
It looks like some tests are redefining symbols by adding the same file already included in Fast DDS library again, could you check it?

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test this

**Comment by richiware** [comment]
@richiprosima please test this

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test windows

**Review by jsantiago-eProsima — APPROVED** [review]

**Pull request merged**